benchmarks/Surrogates/cantilever.jmd

```diff
@@ -0,0 +1,160 @@
+---
+title: Cantilever Beam Function
+author: Mridul Jain, Chris Rackauckas
+---
+
+The Cantilever Beam function is defined as:
+``f(w,t) = \frac{4L^3}{Ewt}*\sqrt{ (\frac{Y}{t^2})^2 + (\frac{X}{w^2})^2 }``
+With parameters L,E,X and Y given.
+
+## Package Imports
+
+```julia
+using Surrogates
+using SurrogatesPolyChaos
+using Plots
+using Statistics
+using PrettyTables
+using BenchmarkTools
+```
+
+## Define the function
+
+```julia
+function canti_lever(x)
+    t = x[1]
+    w = x[2]
+    L = 100.0
+    E = 2.770674127819261e7
+    X = 530.8038576066307
+    Y = 997.8714938733949
+    return (4*L^3)/(E*w*t)*sqrt( (Y/t^2)^2 + (X/w^2)^2)
+end
+```
+
+## Define parameters for training and test data
+
+```julia
+lb = [1.0,1.0]
+ub = [8.0,8.0]
+n_train = 100
+n_test = 150
+```
+
+## Sample training and test data points
+
+```julia
+x_train = sample(n_train, lb, ub, SobolSample())  # Sample training data points
+y_train = canti_lever.(x_train)  # Calculate corresponding function values
+x_test = sample(n_test, lb, ub, RandomSample())  # Sample larger test data set
+y_test = canti_lever.(x_test)  # Calculate corresponding true function values
+```
+
+## Plotting the true function
+  
+```julia
+x, y = 0.0:8.0, 0.0:8.0
+p1 = surface(x, y, (x1,x2) -> canti_lever((x1,x2)))
+xs = [xy[1] for xy in x_train]
+ys = [xy[2] for xy in x_train]
+p2 = contour(x, y, (x1,x2) -> f((x1,x2)))
+scatter!(xs, ys)
+plot(p1, p2, title="True function")
+```
+
+## Fitting the surrogate models
+
+```julia
+mypoly_surrogate = PolynomialChaosSurrogate(xys, zs,  lb, ub)
```

Comment:
This looks incorrect. Shouldn't it be `x_train`, `y_train` instead of `xys`, `zs`?

Reply:
Yeah, will just replace it after testing once